mlflow/models/flavor_backend_registry.py

```diff
@@ -38,7 +38,8 @@ def _get_flavor_backend_for_local_model(model=None, build_docker=True, **kwargs)
 def get_flavor_backend(model_uri, **kwargs):
     if model_uri:
         with TempDir() as tmp:
-            if ModelsArtifactRepository.is_models_uri(model_uri):
+            from mlflow import get_registry_uri
+            if ModelsArtifactRepository.is_models_uri(model_uri) and not is_databricks_unity_catalog_uri(get_registry_uri()):
```

Comment:
AFAICT we're special casing model registry URIs here because the logic below
```
            local_path = _download_artifact_from_uri(
                append_to_uri_path(underlying_model_uri, MLMODEL_FILE_NAME), output_path=tmp.path()
            )
```

May not work for `models:/` URIs (or probably at least didn't when originally written) since URIs like `models:/<modelname>/<version>/path/to/file` weren't supported

Instead of special-casing UC, it'd be better to just call `get_artifact_repository` on `model_uri` to get the artifact repo, then `download_artifacts` similar to https://github.com/mlflow/mlflow/pull/8764/files#diff-c3efddc751d03e91a2284128768d6761b80f9b03ad48947c9f701dbf743b25aaR78